src/setup/node.rs

```diff
@@ -26,47 +26,39 @@ pub struct Node {
 }
 
 impl Node {
-    pub fn new(path: PathBuf, ip_addr: IpAddr) -> Result<Self> {
-        let config = NodeConfig::new(path, ip_addr);
-        let meta = NodeMetaData::new(config.path.clone())?;
-
-        Ok(Self {
-            config,
-            meta,
-            process: None,
-        })
-    }
-
-    pub async fn start(path: PathBuf, ip_addr: IpAddr, peers: Vec<SocketAddr>) -> Result<Self> {
-        let mut node = Self::new(path, ip_addr)?;
-        node.log_to_stdout(false)
-            .initial_peers(peers)
-            .start_process()
-            .await?;
-        Ok(node)
-    }
-
     pub fn addr(&self) -> SocketAddr {
         self.config.local_addr
     }
 
-    pub fn initial_peers(&mut self, addrs: Vec<SocketAddr>) -> &mut Self {
-        self.config.initial_peers = addrs.into_iter().collect();
-        self
-    }
+    // TODO change to consume self, it's probably useless now anyway
```

Comment:
maybe it could be included in the `Drop` impl instead? that one takes `&mut self`

Reply:
`stop` returns `Result`. On one hand I don't want to panic in drop, on the other I don't want to swallow it there. I'm not sure what would be nicest for the future.